### PR TITLE
#287: Use Link-to-Web resource kind from ArkLib

### PR DESCRIPTION
**Pull request by hhio618**
### Description
Related to https://github.com/ark-builders/arklib/issues/10

### CODE REVIEW

File: app/src/main/java/space/taran/arknavigator/mvp/model/repo/preview/generator/LinkPreviewGenerator.kt [hunk 1]
```diff
@@ -18,13 +18,11 @@ object LinkPreviewGenerator : PreviewGenerator() {
...
-        val zip = ZipFile(source.toFile())
-        val entries = zip.entries()
-        val imageEntry = entries
-            .asSequence()
-            .find { entry -> entry.name == IMAGE_FILE }
-            ?: error("No image inside .link file")
-
-        return BitmapFactory.decodeStream(zip.getInputStream(imageEntry))
+        val previewBytes =
+            loadLinkPreview(source.pathString) ?: error("No image inside .link file")
```
Comment (kirillt) [inline app/src/main/java/space/taran/arknavigator/mvp/model/repo/preview/generator/LinkPreviewGenerator.kt:22]: Does `error` produce toast or write to log?

Comment (mdrlzy) [reply app/src/main/java/space/taran/arknavigator/mvp/model/repo/preview/generator/LinkPreviewGenerator.kt:22]: Yes, we [write](https://github.com/ARK-Builders/ARK-Navigator/blob/4b20d31b22da9aa487e217b6b5ad7865bbf35887/app/src/main/java/space/taran/arknavigator/mvp/model/repo/index/PlainResourcesIndex.kt#L229) to log if we failed to generate preview

File: build.gradle [hunk 2]
```diff
@@ -36,6 +36,6 @@ allprojects {
         maven { url "https://plugins.gradle.org/m2/" }
         gradlePluginPortal()
         maven { url 'https://jitpack.io' }
-        maven githubPackage.invoke("ARK-Builders/arklib-android")
+        maven githubPackage.invoke("hhio618/arklib-android")
```
Comment (kirillt) [inline build.gradle:39]: Now we merged arklib-android PR, so this can be adjusted

**Comment by kirillt** [comment]
Issue #287 should be fixed by this PR (it is caused by inconsistency of writer in Shelf and reader in Navigator).
If it is solved indeed, PR #288 will be canceled (which makes reader code tolerable to unknown keys).

**Pull request merged**